src/admin/components/SearchForm.js

```diff
@@ -56,16 +55,16 @@ export class AdminSearchFormBase extends React.Component {
         <input className="AdminSearchForm-query" type="search" name="q"
                ref={(ref) => { this.searchQuery = ref; }}
                placeholder={_('Search')} defaultValue={query} />
-        <button className="button button-middle" type="submit" title="Enter"
+        <Button className="button-middle" type="submit" title="Enter"
                 ref={(ref) => { this.submitButton = ref; }}
-                onClick={this.handleSearch}>
+                onClick={this.handleSearch} styleName="action">
           {_('Search')}
-        </button>
-        <button className="button button-end button-inverse"
+        </Button>
+        <Button className="button-end" inverse styleName="action"
```

Comment:
My immediate reaction to this is I find this less instant than the previous classNames. I wonder if it might be simpler to abstract the `link` vs `button` but otherwise just pass things like className through.

Too many props might make it more complicated to understand e.g.`inverse` prop vs `styleName` etc.